flowmachine/flowmachine/core/hour_slice.py

```diff
@@ -0,0 +1,258 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+from operator import lt as less_than, ge as greater_or_equal
+from typing import List, Union
+
+from sqlalchemy.sql import func, and_, or_, true
+
+__all__ = ["HourInterval", "HourSlice"]
+
+
+class DayPeriod:
+    """
+    Represents a repeated "daily" time period.
+    """
+
+    def __init__(self, weekday=None):
+        self.freq = "day"
+        self.weekday = weekday
+        if self.weekday is not None:
+            raise ValueError(
+                "If freq='day' then the `weekday` argument must not be provided."
+            )
+
+    def filter_timestamp_column_by_day_of_week(self, ts_col):
+        # no additional filtering needed to limit the day of the week
+        return true()
+
+
+class DayOfWeekPeriod:
+    """
+    Represents a repeated "day-of-the-week" time period.
+    """
+
+    valid_weekdays = [
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+        "Sunday",
+    ]
+    weekday_indices_postgres = {
```

Comment:
Could use `isodow` in the extract and then this isn't necessary?

Reply:
How do you mean? I guess I can look up the index in the `valid_weekdays` array instead, but doesn't this also work for `dow`? Or are you suggesting something different?

Reply:
Well, you have to do this odd little dance because Postgres-side, extracting the day of the week gives you 0-6 Sun-Sat, but Postgres also allows you you do `EXTRACT(ISODOW FROM TIMESTAMP '2001-02-18 20:38:40')`, which gives you 1-7, Monday-Sunday, same as `datetime.datetime.strptime(weekday, "%A").isoweekday()`.

Reply:
Ah gotcha! Yes, good point to use the iso weekday to keep indices consistent between postgres and python. 👍

Reply:
Although sadly it turns out the `datetime` module can't just process weekdays without an explicit date. :(
```python
>>> import datetime
>>> print(datetime.datetime.strptime("Monday", "%A").isoweekday())
1
>>> print(datetime.datetime.strptime("Tuesday", "%A").isoweekday())
1
>>> print(datetime.datetime.strptime("Friday", "%A").isoweekday())
1
```

Reply:
Nuts. Well, we do also have `dateutil` as a dependency, so you could do:

```python
from dateutil.parser import parse
parse(weekday).isoweekday()
```

Reply:
That's a nice idea! The minor downside is that `parse()` is very liberal in what it accepts, so you _can_ fool it:
```python
>>> from dateutil.parser import parse
>>> parse("Monday").isoweekday()
1
>>> parse("Monday, 2019-04-25").isoweekday()
4
```
But I can always combine it with an additional `strptime(weekday, "%A")` check...